packages/block-library/src/navigation/index.php

```diff
@@ -67,6 +67,41 @@ function build_css_font_sizes( $attributes ) {
 	return $font_sizes;
 }
 
+/**
+ * Filters out links with no labels.
+ *
+ * @param array $block Block to check label attribute on.
+ *
+ * @return boolean
+ */
+function gutenberg_has_navigation_label( $block ) {
+	return ! empty( $block['attrs']['label'] );
+}
+
+/**
+ * Recursively filters out links with no labels to build a clean navigation block structure.
+ *
+ * @param array $blocks Navigation link inner blocks from the Navigation block.
+ *
+ * @return boolean
+ */
+function gutenberg_remove_empty_navigation_links_recursive( $blocks ) {
+
+	$blocks = array_filter( $blocks, 'gutenberg_has_navigation_label' );
+
+	if ( empty( $blocks ) ) {
```

Comment:
Personal preference: If the conditional was reversed the foreach loop could be moved in its body and the function would have only one return statement.

Reply:
I like early returns in general, but I can go with it :)